cylc/rose/utilities.py

```diff
@@ -325,6 +325,72 @@ def merge_rose_cylc_suite_install_conf(old, new):
     return old
 
 
+def parse_cli_defines(define: str) -> Union[
+    bool, Tuple[
+        List[Union[str, Any]],
+        Union[str, Any],
+        Union[str, Any]
+    ]
+]:
+    """Parse a define string.
+
+    Args:
+        define:
+            A string in one of two forms:
+            - `key = "value"`
+            - `[section]key = "value"`
+
+            With optional `!` and `!!` prepended, indicating an ignored state,
+            which should lead to a warning being logged.
+
+    Returns:
+        False: If state is ignored or trigger-ignored, otherwise...
+        (keys, value, state)
+
+    Examples:
+        # Top level key
+        >>> parse_cli_defines('root-dir = "foo"')
+        (['root-dir'], '"foo"', '')
+
+        # Marked as ignored
+        >>> parse_cli_defines('!root-dir = "foo"')
+        False
+
+        # Inside a section
+        >>> parse_cli_defines('[section]orange = "segment"')
+        (['section', 'orange'], '"segment"', '')
+    """
+    match = re.match(
+        (
+            r'^\[(?P<section>.*)\](?P<state>!{0,2})'
+            r'(?P<key>.*)\s*=\s*(?P<value>.*)'
+        ),
+        define
+    )
+    if match:
+        groupdict = match.groupdict()
+        keys = [groupdict['section'].strip(), groupdict['key'].strip()]
+    else:
+        # Doesn't have a section:
+        match = re.match(
+            r'^(?P<state>!{0,2})(?P<key>.*)\s*=\s*(?P<value>.*)', define)
+        if match:
+            groupdict = match.groupdict()
+            keys = [groupdict['key'].strip()]
+        else:
+            # This seems like it ought to be an error,
+            # But behaviour is consistent with Rose 2019
+            # See:
+            return False
+
+    if match['state'] in ['!', '!!']:
+        LOG.warning(
+            'CLI opts set to ignored or trigger-ignored will be ignored.')
+        return False
+    else:
+        return (keys, match['value'], match['state'])
```

Comment:
I don't think we should be failing for ignored configs, I'm not sure that warning is helpful so we could just:

```suggestion
    return (keys, match['value'], match['state'])
```